charts/device-plugin-operator/templates/operator.yaml

```diff
@@ -427,8 +437,12 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        image: "{{ .Values.image.hub }}/intel-deviceplugin-operator:{{ .Values.image.tag | default .Chart.AppVersion  }}"
-        imagePullPolicy: {{ .Values.image.pullPolicy }}
+      {{- if .Values.privateRegistry.registrySecret }}
+      imagePullSecrets:
+      - name: {{ .Values.privateRegistry.registrySecret }}
+      {{- end }}
+        image: "{{ .Values.manager.image.hub }}/intel-deviceplugin-operator:{{ .Values.manager.image.tag | default .Chart.AppVersion  }}"
```

Comment:
The indentation is wrong here. imagePullSecrets includes container image details like "image"
It should be:
```
imagePullSecrets:
  - name: foobar
containers:
  - name: operator
```

Reply:
Sorry, my example is also wrong. The point I was trying to say is that with the current patch, the container spec would be under the imagePullSecrets. And that's wrong.

Reply:
This section
```
{{- if .Values.privateRegistry.registrySecret }}
imagePullSecrets:
  - name: {{ .Values.privateRegistry.registrySecret }}
{{-end }}
```
should be completely BEFORE "containters" section
e.g.

```
spec:
    {{- if .Values.privateRegistry.registrySecret }}
    imagePullSecrets:
      - name: {{ .Values.privateRegistry.registrySecret }}
    {{- end }}
    containers:
    - args:
      {{- if .Values.controllerExtraArgs }}
      {{- with .Values.controllerExtraArgs }}
      {{- tpl . $ | trim | nindent 8 }}
      {{- end }}
...
```